main.tf

```diff
@@ -47,69 +47,65 @@ module "key_protect" {
 # Key Protect Key Rings
 ##############################################################################
 
-# Create Key Rings included in var.existing_key_map
+locals {
+  key_rings = flatten([
+    for key_ring in var.key_map :
+    key_ring.existing_key_ring ? [] : [{
+      key_ring_name = key_ring.key_ring_name
+      force_delete  = key_ring.force_delete
+    }]
+  ])
+}
+
 module "key_protect_key_rings" {
```

Comment:
`key_protect_key_rings` -> `kms_key_rings` (might need moved block to prevent breaking change